src/ziggurat/metrics.clj

```diff
@@ -1,44 +1,64 @@
-  (let [metric-name (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
-        tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String ["actor" @group]))]
-    (.meter ^MetricRegistry metrics-registry ^MetricName tagged-metric)))
+  ([category metric]
+   (mk-meter category metric nil))
+  ([category metric additional-tags]
+   (let [metric-name   (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
+         tags          (merge-tags additional-tags)
+         tagged-metric (.tagged ^MetricName metric-name tags)]
+     (.meter ^MetricRegistry metrics-registry ^MetricName tagged-metric))))
 
 (defn mk-histogram
-  [category metric]
-  (let [metric-name (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
-        tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String ["actor" @group]))]
-    (.histogram ^MetricRegistry metrics-registry tagged-metric)))
+  ([category metric]
+   (mk-histogram category metric nil))
+  ([category metric additional-tags]
+   (let [metric-name   (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
+         tags          (merge-tags additional-tags)
+         tagged-metric (.tagged ^MetricName metric-name tags)]
+     (.histogram ^MetricRegistry metrics-registry tagged-metric))))
+
+(defn- intercalate-dot
+  [names]
+  (apply str (interpose "." names)))
+
+(defn- inc-or-dec-count
+  [sign metric-namespaces metric additional-tags]
+  (let [metric-namespace (intercalate-dot metric-namespaces)
+        meter            ^Meter (mk-meter metric-namespace metric additional-tags)]
+    (.mark meter (sign 1))))
 
-(defn increment-count
-  ([metric-namespace metric]
-   (increment-count metric-namespace metric 1))
-  ([metric-namespace metric n]
-   (let [meter ^Meter (mk-meter metric-namespace metric)]
-     (.mark meter (int n)))))
+(def increment-count (partial inc-or-dec-count +))
 
-(defn decrement-count
-  ([metric-namespace metric]
-   (decrement-count metric-namespace metric 1))
-  ([metric-namespace metric n]
-   (let [meter ^Meter (mk-meter metric-namespace metric)]
-     (.mark meter (int (- n))))))
+(def decrement-count (partial inc-or-dec-count -))
 
-(defn report-time [metric-namespace time-val]
-  (let [histogram ^Histogram (mk-histogram metric-namespace "all")]
-    (.update histogram (int time-val))))
+(defn report-time
+  ([metric-namespaces time-val]
+   (report-time metric-namespaces time-val nil))
+  ([metric-namespaces time-val additional-tags]
+   (let [default-namespace (last metric-namespaces)
+         metric-namespace  (intercalate-dot metric-namespaces)
+         histogram         ^Histogram (mk-histogram metric-namespace "all" additional-tags)
+         histogram2        ^Histogram (mk-histogram default-namespace "all" additional-tags)]
```

Comment:
I missed this earlier, but there is an inconsistency here, where increment-count or decrement count, push only 1 metric (and are called twice) but report time is called once and pushes 2 metrics, was this intended??

Reply:
yes this is intended, but I got your point. I did some refactoring for this... done